Assets/Scripts/Gameplay/UI/NetworkSimulationUIMediator.cs

```diff
@@ -0,0 +1,240 @@
+
+        Dictionary<string, NetworkScenario> m_Scenarios = new Dictionary<string, NetworkScenario>();
+
+        bool m_Shown;
+        const string k_None = "None";
+        const string k_PauseString = "Pause";
+        const string k_ResumeString = "Resume";
+
+        void Awake()
+        {
+            // initialize connection presets dropdown
+            var optionData = new List<TMP_Dropdown.OptionData>();
+            foreach (var networkSimulatorPreset in NetworkSimulatorPresets.Values)
+            {
+                m_SimulatorPresets[networkSimulatorPreset.Name] = networkSimulatorPreset;
+                optionData.Add(new TMP_Dropdown.OptionData(networkSimulatorPreset.Name));
+            }
+            m_PresetsDropdown.AddOptions(optionData);
+            m_PresetsDropdown.onValueChanged.AddListener(OnPresetChanged);
+
+            // initialize scenario dropdown
+            optionData = new List<TMP_Dropdown.OptionData>();
+
+            // Adding empty scenario
+            optionData.Add(new TMP_Dropdown.OptionData(k_None));
+
+            // Adding ConnectionsCycle scenario
+            var scenarioName = m_ConnectionsCycleScenario.GetType().Name;
+            m_Scenarios[scenarioName] = m_ConnectionsCycleScenario;
+            optionData.Add(new TMP_Dropdown.OptionData(scenarioName));
+
+            // Adding RandomConnectionsSwap scenario
+            scenarioName = m_RandomConnectionsSwapScenario.GetType().Name;
+            m_Scenarios[scenarioName] = m_RandomConnectionsSwapScenario;
+            optionData.Add(new TMP_Dropdown.OptionData(scenarioName));
+
+            m_ScenariosDropdown.AddOptions(optionData);
+            m_ScenariosDropdown.onValueChanged.AddListener(OnScenarioChanged);
+
+            // Hide UI until ready
+            Hide();
+        }
+
+        void Start()
+        {
+            NetworkManager.Singleton.OnClientStarted += OnNetworkManagerStarted;
+            NetworkManager.Singleton.OnServerStarted += OnNetworkManagerStarted;
+        }
+
+        void OnDestroy()
+        {
+            if (NetworkManager.Singleton is not null)
+            {
+                NetworkManager.Singleton.OnClientStarted -= OnNetworkManagerStarted;
+                NetworkManager.Singleton.OnServerStarted -= OnNetworkManagerStarted;
+            }
+        }
+
+        void OnNetworkManagerStarted()
+        {
+            if (m_NetworkSimulator.IsAvailable)
+            {
+                Show();
+            }
+        }
+
+        void OnPresetChanged(int optionIndex)
+        {
+            m_NetworkSimulator.ChangeConnectionPreset(m_SimulatorPresets[m_PresetsDropdown.options[optionIndex].text]);
+        }
+
+        void OnScenarioChanged(int optionIndex)
+        {
+            var scenarioName = m_ScenariosDropdown.options[optionIndex].text;
+            m_NetworkSimulator.Scenario = m_Scenarios.ContainsKey(scenarioName) ? m_Scenarios[scenarioName] : null;
```

Comment:
I'm easily able to reproduce an error when toggling a new scenario when ConnectionCycle scenario is already active.
Reproduction steps:
* Hit ConnectionsCycle scenario from dropdown
* While scenario is running, select another scenario from dropdown that isn't None.
* While new scenario is running, select ConnectionsCycle scenario once again.

Reply:
Coming back to this, not yet sure if this is a Tools bug, or a bad usage from my part. I'll investigate this further

Reply:
Will re-review once more when we find out if it's on our end or not, so I'll keep this open for now.